### DISCUSSION TITLE
extUeAppMecApp emulation with several base stations

### DISCUSSION THREAD

**Discussion by hcr-git** [Unipisa/Simu5G#226] [Q&A]
Hello,

I would like to ask about the emulation mode of Simu5G.

We are enhancing the extUeAppMecApp emulation by adding a few more base stations.

Currently, we are connecting the additional base stations to the iUpf element and assigning a different interface to the iUpf element to the new gnbs, for example gnb2:

(excerpt of iUpf.mrt - added)
# interface to the gnb2
name: ppp3
    inet_addr: 10.0.6.1
    Mask: 255.255.255.0
    MTU: 1500
    POINTTOPOINT MULTICAST

However, we are encountering difficulties with updating the routing tables of Simu5G elements, namely the routes in iUpf.mrt. 

The eu.mrt file assigns the 10.0.0.1 address to the UE/car element. How should it work when the ue/car moves to another base station? It should acquire a new address in the cellular network defined by us in gbn2.mrt?

Thank you in advance,
Helena Rodrigues

**Comment by giovanninardini** [comment]
Hello,

in simu5G is is assumed that all UEs and gNBs belong to the same “local wireless network”. This means that the UE does not need to change its IP address when moving to another base station. The correct routing in the core network is handled by the UPF (the entry point of the 5G network), which queries the Binder to know which gNB the destination UE is attached to, and tunnels the packet to that gNB using GTP.
As far as the iUPF is concerned, it should not even need to know the route to the IP address of the UE, as it just handles tunneled packets (whose dest IP address is the one of the gNB).

Hope this helps.

Best regards.
Giovanni
